pkg/catalog/catalog.go

```diff
@@ -1711,6 +1717,99 @@ func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repositoryID string
 	return c.Store.SaveGarbageCollectionCommits(ctx, repository, previousRunID)
 }
 
+func createParquetFile(entries []DBEntry) (string, error) {
+	// TODO: Modify using a run ID generator (https://github.com/treeverse/lakeFS/issues/4469)
+	filename := uuid.New().String()
+	fw, err := local.NewLocalFileWriter(filename)
+	if err != nil {
+		return "", err
+	}
+	defer fw.Close()
+	pw, err := writer.NewParquetWriter(fw, new(DBEntry), int64(len(entries)))
+	if err != nil {
+		return "", err
+	}
+	// Compression type
+	pw.CompressionType = parquet.CompressionCodec_GZIP
+	for _, entry := range entries {
+		if err = pw.Write(entry); err != nil {
+			return "", err
+		}
+	}
+	if err = pw.WriteStop(); err != nil {
+		return "", err
+	}
+	_, err = fw.Seek(0, 0)
+	if err != nil {
+		return "", err
+	}
+	return filename, nil
+}
+
+func markToString(mark *graveler.GCUncommittedMark) (*string, error) {
+	if mark == nil {
+		return nil, nil
+	}
+	var buf bytes.Buffer
+	encoder := base64.NewEncoder(base64.StdEncoding, &buf)
+	err := json.NewEncoder(encoder).Encode(mark)
+	if err != nil {
+		return nil, err
+	}
+	result := buf.String()
+	return &result, nil
+}
+
+func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID, runID, contToken string) (*graveler.GarbageCollectionRunMetadata, *string, error) {
```

Comment:
This is a partial implementation according to the plan. The write of the `copies table` is missing. How would we paginate that?

Reply:
2 options:
1. graveler GetGarbageCollectionUncommitted once we reach the end we can safely read the copy table as part of the data
2. In catalog perform another call in graveler to explicitly read the table